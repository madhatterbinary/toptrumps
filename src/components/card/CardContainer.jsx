import React from 'react';
import PropTypes from 'prop-types';
import Card from './Card';
import SelectedAttribute from './SelectedAttribute';

const CardContainer = (props) => {
  const {
    card,
    isClosed,
    isWinner,
    selectedAttribute,
    handleAttributeSelection,
    status,
  } = props;

  return (
    <div className={`card-container ${isWinner && 'winner'}`}>
      <Card 
        card={card}
        isClosed={isClosed}
        handleAttributeSelection={handleAttributeSelection}
        status={status}
        selectedAttribute={selectedAttribute}
      />
      {!isClosed ? (
      <SelectedAttribute
        selectedAttribute={selectedAttribute}
        attributeValue={card[selectedAttribute]}
        isClosed
        isWinner={isWinner}
      />
      ) : null}
      <div className="card-result">You won!</div>
    </div>
  );
}

CardContainer.propTypes = {
  card: PropTypes.object.isRequired,
  isClosed: PropTypes.bool.isRequired,
  isWinner: PropTypes.bool.isRequired,
  status: PropTypes.string.isRequired,
  selectedAttribute: PropTypes.string,
  handleAttributeSelection: PropTypes.func.isRequired,
}

export default CardContainer;
