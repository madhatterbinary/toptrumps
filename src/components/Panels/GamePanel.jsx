import React from 'react';
import PropTypes from 'prop-types';
import CardContainer from '../card/CardContainer';
import {
  STATUS_READY,
  STATUS_OVER,
  PLAYER_1,
  PLAYER_2,
} from '../../constants/constants';

const GamePanel = (props) => {
  const {
    player1Card,
    player2Card,
    status,
    handleAttributeSelection,
    selectedAttribute,
    lastWinner,
    isDraw,
  } = props;

  return (
    <div className="game-panel">
      { status === STATUS_OVER && (
        <div className="game-nav-result">Game Over</div>
      )}
      { isDraw && (
        <div className="game-nav-result">Draw</div>
      )}
      <CardContainer
        card={player1Card}
        isClosed={status === STATUS_READY && lastWinner === PLAYER_2}
        handleAttributeSelection={handleAttributeSelection}
        selectedAttribute={selectedAttribute}
        status={status}
        isWinner={status !== STATUS_READY && lastWinner === PLAYER_1 && !isDraw}
      />
      <CardContainer 
        card={player2Card}
        isClosed={status === STATUS_READY && lastWinner !== PLAYER_2}
        handleAttributeSelection={handleAttributeSelection}
        selectedAttribute={selectedAttribute}
        status={status}
        isWinner={status !== STATUS_READY && lastWinner === PLAYER_2 && !isDraw}
      />
    </div>
  );
}

GamePanel.propTypes = {
  status: PropTypes.string.isRequired,
  player1Card: PropTypes.object,
  player2Card: PropTypes.object,
  lastWinner: PropTypes.string,
  selectedAttribute: PropTypes.string,
  handleAttributeSelection: PropTypes.func.isRequired,
}

export default GamePanel;
