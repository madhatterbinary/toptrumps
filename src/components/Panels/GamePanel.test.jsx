import React from 'react';
import GamePanel from './GamePanel';
import CardContainer from '../card/CardContainer';

import TestRenderer from 'react-test-renderer';

describe('GamePanel', () => {
  it('renders', () => {
    const component = TestRenderer.create(<GamePanel player1Card={{}} player2Card={{}} handleAttributeSelection={()=>{}} lastWinner="" selectedAttribute="" status="" isDraw={false} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(component.root.findAllByType(CardContainer)).not.toBeNull();
  });
});
