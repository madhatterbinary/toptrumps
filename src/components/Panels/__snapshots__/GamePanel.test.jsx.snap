// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GamePanel renders 1`] = `
<div
  className="game-panel"
>
  <div
    className="card-container false"
  >
    <div
      className="card false"
    >
      <div
        className="card-body"
      >
        <div
          className="card-header"
        >
          <h3 />
          <div
            className="card-additional-info"
          >
            <div>
              Model: 
            </div>
            <div>
              Manufacturer: 
            </div>
          </div>
        </div>
        <div
          className="card-image-container"
        >
          <img
            alt=""
            className="card-image"
          />
        </div>
        <div
          className="card-attribute-list"
        >
          <div
            className="card-attribute-row"
            onClick={[Function]}
          >
            <span
              className="card-attribute-label false"
            >
              Cost
              :
            </span>
            <span
              className="card-attribute-value false"
            >
              
            </span>
          </div>
          <div
            className="card-attribute-row"
            onClick={[Function]}
          >
            <span
              className="card-attribute-label false"
            >
              Length
              :
            </span>
            <span
              className="card-attribute-value false"
            >
              
            </span>
          </div>
          <div
            className="card-attribute-row"
            onClick={[Function]}
          >
            <span
              className="card-attribute-label false"
            >
              Speed
              :
            </span>
            <span
              className="card-attribute-value false"
            >
              
            </span>
          </div>
          <div
            className="card-attribute-row"
            onClick={[Function]}
          >
            <span
              className="card-attribute-label false"
            >
              Crew
              :
            </span>
            <span
              className="card-attribute-value false"
            >
              
            </span>
          </div>
          <div
            className="card-attribute-row"
            onClick={[Function]}
          >
            <span
              className="card-attribute-label false"
            >
              Passengers
              :
            </span>
            <span
              className="card-attribute-value false"
            >
              
            </span>
          </div>
          <div
            className="card-attribute-row"
            onClick={[Function]}
          >
            <span
              className="card-attribute-label false"
            >
              Cargo Capacity
              :
            </span>
            <span
              className="card-attribute-value false"
            >
              
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="selected-attribute"
    >
      Select an attribute
    </div>
    <div
      className="card-result"
    >
      You won!
    </div>
  </div>
  <div
    className="card-container false"
  >
    <div
      className="card false"
    >
      <div
        className="card-body"
      >
        <div
          className="card-header"
        >
          <h3 />
          <div
            className="card-additional-info"
          >
            <div>
              Model: 
            </div>
            <div>
              Manufacturer: 
            </div>
          </div>
        </div>
        <div
          className="card-image-container"
        >
          <img
            alt=""
            className="card-image"
          />
        </div>
        <div
          className="card-attribute-list"
        >
          <div
            className="card-attribute-row"
            onClick={[Function]}
          >
            <span
              className="card-attribute-label false"
            >
              Cost
              :
            </span>
            <span
              className="card-attribute-value false"
            >
              
            </span>
          </div>
          <div
            className="card-attribute-row"
            onClick={[Function]}
          >
            <span
              className="card-attribute-label false"
            >
              Length
              :
            </span>
            <span
              className="card-attribute-value false"
            >
              
            </span>
          </div>
          <div
            className="card-attribute-row"
            onClick={[Function]}
          >
            <span
              className="card-attribute-label false"
            >
              Speed
              :
            </span>
            <span
              className="card-attribute-value false"
            >
              
            </span>
          </div>
          <div
            className="card-attribute-row"
            onClick={[Function]}
          >
            <span
              className="card-attribute-label false"
            >
              Crew
              :
            </span>
            <span
              className="card-attribute-value false"
            >
              
            </span>
          </div>
          <div
            className="card-attribute-row"
            onClick={[Function]}
          >
            <span
              className="card-attribute-label false"
            >
              Passengers
              :
            </span>
            <span
              className="card-attribute-value false"
            >
              
            </span>
          </div>
          <div
            className="card-attribute-row"
            onClick={[Function]}
          >
            <span
              className="card-attribute-label false"
            >
              Cargo Capacity
              :
            </span>
            <span
              className="card-attribute-value false"
            >
              
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="selected-attribute"
    >
      Select an attribute
    </div>
    <div
      className="card-result"
    >
      You won!
    </div>
  </div>
</div>
`;
